src/RewardsStreamerMP.sol

```diff
@@ -411,20 +407,21 @@ contract RewardsStreamerMP is
 
     function _updateVault(address vaultAddress, bool forceMPUpdate) internal {
         VaultData storage vault = vaultData[vaultAddress];
+
+        // first accue pending rewards for the work done so far
+        uint256 rewardsAccrued = _vaultPendingRewards(vault);
+        vault.rewardsAccrued += rewardsAccrued;
+        vault.rewardIndex = lastRewardIndex;
+
+        // then accrue pending MPs
         uint256 accruedMP = _vaultPendingMP(vault);
         if (accruedMP > 0 || forceMPUpdate) {
             vault.mpAccrued += accruedMP;
             vault.lastMPUpdateTime = block.timestamp;
+            totalMPStaked += accruedMP;
         }
 
-        uint256 rewardsAccrued = _vaultPendingRewards(vault);
-        vault.rewardsAccrued += rewardsAccrued;
-        vault.rewardIndex = lastRewardIndex;
-
-        uint256 mpToStake = vault.mpAccrued - vault.mpStaked;
-        vault.mpStaked += mpToStake;
-        totalMPStaked += mpToStake;
-        emit Compound(vaultAddress, mpToStake);
+        emit Compound(vaultAddress, accruedMP);
```

Comment:
@gravityblast @3esmit should we change this event to `vaultUpdated(vaultAddress, rewardsAccrued)` ?

Reply:
@0x-r4bbit yes I agree it's better with the new names